core/src/main/java/com/orientechnologies/orient/core/index/OIndexManagerShared.java

```diff
@@ -521,7 +511,7 @@ public void run() {
         }
 
         try {
-          recreateIndexes(storage, indexesToRebuild, newDb);
+          recreateIndexes(indexesToRebuild, newDb);
         } finally {
           if (storage instanceof OAbstractPaginatedStorage) {

```

Comment:
Seems like `OIndexAbstract#loadFromConfiguration` may rebuild indexes on its own. Not sure is it ok to turn the WAL on while it is doing the reindexing. WDYT?